### ISSUE TITLE
"Class has no attribute property_name" caused by any error in property.

### ISSUE THREAD

**Issue by laundmo** [pallets/jinja#1317]
Whenever a property thats used in a jinja2 template throws any error, jinja2 will replace that error with "Class has no attribute property_name", which is a plain wrong error message.

How to reproduce:
```py
from jinja2 import Environment
from jinja2.runtime import StrictUndefined

class MyClass():
    @property
    def faulty_property(self):
        return self.doesntexist

env = Environment(undefined=StrictUndefined) 
t = env.from_string("{{ obj.faulty_property }}")
print(t.render(obj=MyClass()))
```
Actual error:  `'__main__.MyClass object' has no attribute 'faulty_property'`
Expected error: `'MyClass' object has no attribute 'doesntexist'`

I have implemented a workaround for this myself, the code for which can be found here: https://gist.github.com/laundmo/acb9e26345c1454dd6029391e895f844

Environment:
- Python version: 3.9.0
- Jinja version: Jinja2-2.11.2

**Comment by jab** [comment]
Thanks for the bug report. Feel free to submit a PR! Would definitely be more helpful than sending us a workaround called "Jinja is dumb" 😛

**Comment by davidism** [comment]
Duplicate of #854, see also #313.

**Comment by laundmo** [comment]
> Thanks for the bug report. Feel free to submit a PR! Would definitely be more helpful than sending us a workaround called "Jinja is dumb" 😛

i definitely planned to look into making a PR. sorry the name was just my way to let out the frustration of being held up by this for a few hours, doesn't look that great to others, i admit. to clarify i work with jinja of my own volition and dont think its a bad tool at all.

**Comment by laundmo** [comment]
@davidism shouldn't one of the issues still be open, since this is a actual issue that still persists? both issues you linked, and #604 are closed...

**Comment by laundmo** [comment]
just gonna ask here before i spend a bunch of time looking for it: could anyone point me at the location where the current version of Undefined is instantiated after a error was thrown?

**Comment by davidism** [comment]
This is an issue with Python in general. It's showing the exact same error that Python itself would. I'm not convinced Jinja needs special handling for it.
